ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,295 @@
+import boto.exception
+
+# Cloudify imports
+from ec2 import constants
+from ec2 import connection
+from ec2 import utils
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+
+@operation
+def creation_validation(**_):
+    """ This checks that all user supplied info is valid """
+
+    for property_key in constants.ELB_REQUIRED_PROPERTIES:
+        utils.validate_node_property(property_key, ctx.node.properties)
+
+    if not ctx.node.properties['resource_id']:
+        elb = None
+    else:
+        elb = _get_existing_elb(
+            ctx.node.properties['resource_id'])
+
+    if ctx.node.properties['use_external_resource'] and not elb:
+        raise NonRecoverableError(
+            'External resource, but the supplied '
+            'elb does not exist in the account.')
+
+    if not ctx.node.properties['use_external_resource'] and elb:
+        raise NonRecoverableError(
+            'Not external resource, but the supplied '
+            'elb exists.')
+
+
+@operation
+def create_elb(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        return
+
+    lb = _create_elb()
+
+    health_checks = ctx.node.properties.get('health_checks')
+
+    if isinstance(health_checks, list):
+        if health_checks:
+            for health_check in health_checks:
+                _add_health_check_to_elb(lb, health_check)
+    else:
+        raise NonRecoverableError(
+            'Health Checks not defined properly in inputs. '
+            'Health Checks should be a list.')
+
+
+def _add_instance_to_elb_list_in_properties(instance_id):
+
+    if 'instance_list' not in ctx.target.instance.runtime_properties.keys():
+        ctx.target.instance.runtime_properties['instance_list'] = []
+
+    ctx.target.instance.runtime_properties['instance_list'].append(instance_id)
+
+
+def _remove_instance_from_elb_list_in_properties(instance_id):
+    if instance_id in ctx.target.instance.runtime_properties['instance_list']:
+        ctx.target.instance.runtime_properties['instance_list'].remove(
+            instance_id)
+
+
+@operation
+def remove_instance_from_elb(**_):
+
+    ctx.logger.info('Attempting to remove_instance_from_elb.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    instance_list = [instance_id]
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.deregister_instances(instance_list)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Instance not removed from Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} removed from Load Balancer {1}.'
```

Comment:
Let's do this as a standard:

when starting to perform some kind of operation:

`ctx.logger.info('doing x on y with z...')`

when the operation is complete, there's no need to write the y and z again.. you can just write:

`ctx.logger.info('finished doing x')`